xbmc/filesystem/CurlFile.cpp

```diff
@@ -691,12 +691,29 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
   // set CA bundle file
   std::string caCert = CSpecialProtocol::TranslatePath(
       CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_caTrustFile);
-#ifdef TARGET_WINDOWS_STORE
-  // UWP Curl - Setting CURLOPT_CAINFO with a valid cacert file path is required for UWP
-  g_curlInterface.easy_setopt(h, CURLOPT_CAINFO, "system\\certs\\cacert.pem");
-#endif
   if (!caCert.empty() && XFILE::CFile::Exists(caCert))
     g_curlInterface.easy_setopt(h, CURLOPT_CAINFO, caCert.c_str());
+
+#ifdef TARGET_WINDOWS_STORE
+  // From OpenSSL 3.0 CURLOPT_CAINFO not works (on UWP), use CURLOPT_CAINFO_BLOB instead
+  static std::vector<uint8_t> cachedBuffer;
+  static bool bufferLoaded = false;
+  if (!bufferLoaded)
+  {
+    XFILE::CFile file;
+    if (file.LoadFile(CSpecialProtocol::TranslatePath("special://xbmc/system/certs/cacert.pem"),
+                      cachedBuffer))
+    {
+      bufferLoaded = true;
+    }
+  }
```

Comment:
```suggestion
static std::vector<uint8_t> cachedBuffer;
static std::once_flag loadBufferFlag;
std::call_once(loadBufferFlag, []() {
  XFILE::CFile file;
  file.LoadFile(CSpecialProtocol::TranslatePath("special://xbmc/system/certs/cacert.pem"), cachedBuffer);
});
```
This is pretty lightweight in terms of locking overhead but it does not retry fetching the certificate if the first call fails.

Reply:
Thanks but I am not big fan of use these modernish  c++ extensions for several reasons: now is tendence but in 10 years maybe deprecated, not complete known side effects or contraindications, etc.

I replaced with standard code, in this way is removed #ifdef TAGET_WINDOWS_STORE and cached blob can be reused in a future on other platforms or other uses, etc.